html/tests/masthead.tests.js

```diff
@@ -12,15 +16,113 @@ import {
 import { dropdowns } from '../components/dropdown';
 
 describe('masthead init', () => {
+  let main;
+  let nav;
+  let mastheadDiv;
+  let iconContainer;
+  let iconContainerDiv;
+
+  beforeEach(() => {
+    main = document.createElement('div');
+    main.setAttribute('data-sprk-main', null);
+
+    // Create the main Masthead element
+    mastheadDiv = document.createElement('div');
+    mastheadDiv.classList.add('sprk-c-Masthead');
+    mastheadDiv.setAttribute('data-sprk-masthead', null);
+
+    // Create narrow nav
+    nav = document.createElement('nav');
+    nav.classList.add('sprk-u-Display--none');
+    nav.classList.add('sprk-c-Masthead__narrow-nav');
+    nav.setAttribute('data-sprk-mobile-nav', 'mobileNav');
+
+    // Add nav to masthead
+    mastheadDiv.appendChild(nav);
+
+    // Create a menu button container for icon
+    iconContainer = document.createElement('button');
+    iconContainer.setAttribute('data-sprk-mobile-nav-trigger', 'mobileNav');
+
+    // Create a container div for icon
+    iconContainerDiv = document.createElement('div');
+    iconContainerDiv.classList.add('sprk-c-Masthead__menu');
+
+    // Add iconContainer to iconContainerDiv
+    iconContainerDiv.appendChild(iconContainer);
+
+    // Add iconContainerDiv to mastheadDiv
+    mastheadDiv.appendChild(iconContainerDiv);
+    main.appendChild(mastheadDiv);
+    document.body.appendChild(main);
+  })
+
   afterEach(() => {
-    document.querySelectorAll.restore();
+    if (document.querySelectorAll.restore) {
```

Comment:
What’s the reason for  this new if ?

Reply:
The `.restore` is cleanup from the sinon spy on line 69. Before my task, that test was the only test in that section. When I added my new tests, the `.restore` line started breaking because I wasn't creating a sinon spy that needed to be cleaned up.

Reply:
Thank you! That makes sense!